app/src/pages/Application.tsx

```diff
@@ -0,0 +1,314 @@
+
+  //Function for deciding the different values in the form
+  const { register, handleSubmit, formState: { errors } } = useForm<NewApplication>();
+
+  //Variables for toggling the different fields' visibility
+  const [toggleMotivation, setToggleMotivation] = useState(true);
+  const [toggleAcademicExperiences, setToggleAcademicExperiences] = useState(false);
+  const [toggleProfessionalExperience, setToggleProfessionalExperience] = useState(false);
+
+  /**
+    * OnSubmit function for Application.
+    * Takes the submitted data from the form and sends it to the backend through a service.
+    * Navigates to the Login page upon recieving a succesfull response
+    *
+    * @param {JSON} data Which includes the value of the various fields in the application
+    */
+  const onSubmit: SubmitHandler<NewApplication> = async (data) => {
+    AuthService.postNewApplication({
+      motivation: data.motivation, academicLevel: data.academicLevel, academicStatus: data.academicStatus,
+      major: data.major, institution: data.institution, educationStartDate: data.educationStartDate,
+      educationEndDate: data.educationEndDate, company: data.company, position: data.position, 
+      workStartDate: data.workStartDate, workEndDate: data.workEndDate, workActivities: data.workActivities,
+
+      baseUser: id
+    }).then((res) =>{
+      if(res.status == 201){
+        navigate("/login")
+      }
+    })
+  };
+
+  //Variable for keeping track of the length of the motivation
+  const [motivation, setMotivation] = useState('');
+  const maxLength = 800;
+
+  // Function to make sure motivation is not above 800 characters
+  const handleMotivationChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
+    const text = event.target.value;
+    if (text.length <= maxLength) {
+      setMotivation(text);
+    }
+  };
+
+  const [major, setMajor] = useState('');
+  const [institution, setInstitution] = useState('');
+  const [educationStartDate, setEducationStartDate] = useState('');
+  const [educationEndDate, setEducationEndDate] = useState('');
+  const [company, setCompany] = useState('');
+  const [position, setPosition] = useState('');
+  const [workStartDate, setWorkStartDate] = useState('');
+  const [workEndDate, setWorkEndDate] = useState('');
+  const [workActivities, setWorkActivities] = useState('');
+    
+  function SubmitButton(){
+    if (major){
+      return <button type="submit" className="w-[238px] h-[52px] px-10 py-4 bg-cyan-800 hover:bg-cyan-900 rounded-lg justify-center items-start gap-2.5 inline-flex text-center text-white text-base font-bold font-['Montserrat']">
+              Enviar para análise {/* Send for analysis */}
+             </button>
+    } else {
+      return <button type="submit" className="w-[238px] h-[52px] px-10 py-4 bg-cyan-800 hover:bg-cyan-900 rounded-lg justify-center items-start gap-2.5 inline-flex text-center text-white text-base font-bold font-['Montserrat']" disabled>
```

Comment:
This button could be made into a component perhaps. More importantly, let's just set `disabled=major` in this case and we'd have the same logic without this whole function.

Reply:
Yessir, we originally had it as three components, but we were unsure of how to use `form` spanning three components and a main page. So unfortunately, it ended up being a long unreadable page. But rest assured, it is something we are very much aware of.

Reply:
Instead of the form submit you could probably just use hooks if the form can't see within the components, although I'd think it can